c/zss.c

```diff
@@ -793,18 +822,22 @@ static WebPluginListElt* readWebPluginDefinitions(HttpServer *server, ShortLived
                     if (server->loggingIdsByName != NULL) {
                       if (plugin->dataServiceCount > 0) {
                         for (int i = 0; i < plugin->dataServiceCount; i++) {
-                          size_t keyLength = strlen(plugin->dataServices[i]->identifier);
-                          char key[keyLength+1];
-                          memset(&key, 0, sizeof(key));
-                          strncpy(key, plugin->dataServices[i]->identifier, keyLength);
-                          for (int j = 0; j < keyLength; j++) {
-                            if (key[j] == '/') {
-                              key[j] = ':';
-                            }
+                          size_t identifierLength = strlen(plugin->dataServices[i]->identifier) + 1;
+                          char *key = formatDataServiceIdentifier(plugin->dataServices[i]->identifier, 
+                                                                  identifierLength, ':');
+                          if (key) {
+                            htPut(server->loggingIdsByName,
+                                  key,
+                                  &(plugin->dataServices[i]->loggingIdentifier));
+                            safeFree(key, identifierLength);
```

Comment:
Actually, `safeFree` is not needed here because the key is  in the `hashtable`.

Reply:
Resolved, revmoved safeFree().